ashley/src/com/badlogic/ashley/signals/Signal.java

```diff
@@ -56,9 +56,8 @@ public void removeAllListeners () {
 	 * @param object The object to send off
 	 */
 	public void dispatch (T object) {
-		Object[] items = listeners.begin();
-		for (int i = 0, n = listeners.size; i < n; i++) {
-			Listener<T> listener = (Listener<T>)items[i];
+		final Listener<T>[] items = listeners.begin();
```

Comment:
Arrays of generics is a bad idea and should generally be avoided. In this case, it looks like the base libgdx library is returning it, however, so there isn't much that can be done here.

See: http://www.angelikalanger.com/Articles/Papers/JavaGenerics/ArraysInJavaGenerics.htm (specifically the Parameterized Types As Array Elements section), if you are interested :).